app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -622,7 +622,25 @@ class NewMessageViewModel @Inject constructor(
     private fun isSavingDraftWithoutChanges(action: DraftAction) = action == DraftAction.SAVE && snapshot?.hasChanges() != true
 
     fun updateIsSendingAllowed() {
-        isSendingAllowed.value = draft.to.isNotEmpty() || draft.cc.isNotEmpty() || draft.bcc.isNotEmpty()
+
+        val hasRecipient = draft.to.isNotEmpty() || draft.cc.isNotEmpty() || draft.bcc.isNotEmpty()
+        if (!hasRecipient) {
+            isSendingAllowed.value = false
+            return
+        }
+
+        var size = 0L
+        var isSizeCorrect = true
+        run breaking@{
+            draft.attachments.forEach {
```

Comment:
We could do a `sumBy` even though we'll iterate through the whole array but it'll be much easier to read.

Reply:
True, but we don't want to iterate through the whole array.